rocketpy/motors/Tank.py

```diff
@@ -141,19 +199,19 @@ def liquidCenterOfMass(self):
             Center of mass of the liquid portion of the tank as a
             function of time.
         """
+        balance = self.geometry.balance(self.geometry.bottom, self.liquidHeight.max)
+        liquid_balance = balance @ self.liquidHeight
+        centroid = liquid_balance / self.liquidVolume
 
-        def evaluate_liquid_com(t):
-            mass_integrand = Function(
-                lambda h: h * self.liquid.density * self.geometry.area(h)
-            )
-            if self.liquidMass(t) > 1e-6:
-                return mass_integrand.integral(
-                    self.geometry.bottom, self.liquidHeight(t)
-                ) / self.liquidMass(t)
-            else:
-                return self.geometry.bottom
+        # Check for zero liquid volume
+        bound_volume = self.liquidVolume < 1e-4 * self.geometry.total_volume
+        if bound_volume.any():
+            # TODO: pending Function setter impl.
+            centroid.yArray[bound_volume] = self.geometry.bottom
+            centroid.setInterpolation()
+            centroid.setExtrapolation()
```

Comment:
I wish there was a way around doing this, but let's leave it for our future selves to find it.
How did you decide that `1e-4` was a good factor to check for in order to prevent numerical errors causing the centroid to go crazy?

Reply:
The reasoning behind the decision was more practical than theoretical, unfortunately I was not able to pinpoint a tolerance so that it would always be free of numerical instabilities. Therefore the rationale was the practical implication of a tank with less than `0,1%` of its own volume in fluid: the centroid is at the bottom.

There is nothing specific about this number and I am really not fond of hardcoded tolerance values. As you said, this is an issue for the future, for now it works well enough.